pkg/azureutils/azure_disk_utils.go

```diff
@@ -727,11 +754,19 @@ func GetMaxSharesAndMaxMountReplicaCount(parameters map[string]string) (int, int
 		klog.Warningf("maxShares cannot be set smaller than 1... Defaulting current maxShares (%d) value to 1", maxShares)
 		maxShares = 1
 	}
-	if maxShares-1 < maxMountReplicaCount {
-		klog.Warningf("maxMountReplicaCount cannot be set larger than maxShares - 1... Defaulting current maxMountReplicaCount (%d) value to (%d)", maxMountReplicaCount, maxShares-1)
-		maxMountReplicaCount = maxShares - 1
-	} else if maxMountReplicaCount < 0 {
-		maxMountReplicaCount = maxShares - 1
+
+	if isMultiNodeVolume {
+		if maxMountReplicaCount > 0 {
+			klog.Warning("maxMountReplicaCount is ignored for volumes that can be mounted to multiple nodes... Defaulting current maxMountReplicaCount (%d) to 0", maxMountReplicaCount)
+		}
+		maxMountReplicaCount = 0
```

Comment:
Should we return here instead of having an `else` statement below.

Reply:
Sure.